readLog/src/main/java/org/corfudb/readLog/readLog.java

```diff
@@ -0,0 +1,354 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.Map;
+
+public class readLog {
+    static int MetadataSize;
+    static int LogHeaderSize;
+    static final String USAGE =
+            "Usage:\n"
+                    + "\treadLog [-d <show binary> | -e <address> | -T <address>] <log_file>"
+                    + "\n"
+                    + "Options:\n"
+                    + " -d <show binary>, --display=<show binary>   Display log records (include binary data)\n"
+                    + " -e <address>, --erase=<address>             Erase log record\n"
+                    + " -T <address>, --erase_tail=<address>        Erase log records to end of file\n";
+    static final short DELIMITER = 0x4c45;
+
+    public static void main(String[] args) {
+        readLog reader = new readLog();
+        reader.run(args);
+        reader.cleanUp();
+    }
+
+    public readLog() {
+        fileStreamIn = null;
+        fileStreamOut = null;
+    }
+
+    public int run(String[] args) {
+        try {
+            boolean ret = init(args);
+        } catch(DocoptExitException e) {
+            System.out.println(USAGE);
+            System.exit(1);
+        }
+        return readAll();
+    }
+
+    public boolean init(String[] args) {
+        Docopt parser = new Docopt(USAGE);
+        parser.withExit(false);
+        Map<String, Object> opts = parser.parse(args);
+        op = new Operation(Operation.OperationType.REPORT, 0);
+        String log_file_name = new String();
+        boolean use_output_file = false;
+        if (opts.get("<log_file>") != null) {
+            log_file_name = (String) opts.get("<log_file>");
+            System.out.println("Log file: " + log_file_name);
+        } else {
+            System.out.print(USAGE);
+            System.out.println("Return FALSE");
+            return false;
+        }
+        if (opts.get("--erase") != null) {
+            int addr = Integer.parseInt(((String) opts.get("--erase")).trim());
+            op = new Operation(Operation.OperationType.ERASE_ONE, addr);
+            use_output_file = true;
+        } else if (opts.get("--erase_tail") != null) {
+            int addr = Integer.parseInt(((String) opts.get("--erase_tail")).trim());
+            op = new Operation(Operation.OperationType.ERASE_TAIL, addr);
+            use_output_file = true;
+        } else if (opts.get("--display") != null) {
+            int flag = Integer.parseInt(((String) opts.get("--display")).trim());
+            if (flag == 0) {
+                op = new Operation(Operation.OperationType.DISPLAY, 0);
+            } else {
+                op = new Operation(Operation.OperationType.DISPLAY_ALL, 0);
+            }
+        }
+        LogHeader lh = LogHeader.newBuilder()
+                .setVersion(10)
+                .setVerifyChecksum(true)
+                .build();
+        LogHeaderSize = lh.getSerializedSize();
```

Comment:
This might break. Protobuf uses variable length encoding, so depending on the integer range, so the number of encoding bytes my changed as the version is incremented. 

The LogHeader is prefixed with a Metadata message (that has a fixed length), from the metadata message you can extract the log header size.